utils/sl/comp_build.sh

```diff
@@ -0,0 +1,405 @@
+  job_end=${JOB_SUFFIX}${job_matrix}
+
+  #Use the latest by default
+  declare $upper_name=${CORAL_ARTIFACTS}/${name}${job_end}/latest
+  latest=
+  if [ "${JOB_NAME}" != "${B_COMP}${JOB_SUFFIX}" ]; then
+    #If it's not the scratch job, use a previous build of the scratch job
+    echo "Getting $name build from $B_COMP latest"
+    comp_name=${B_COMP}
+    latest=$(readlink -f ${CORAL_ARTIFACTS}/${B_COMP}${job_end}/latest)
+  elif [ -n "$cname" ]; then
+    #If it is the scratch job, use a child job, if specified
+    echo "Getting $name build from $cname latest"
+    latest=$(readlink -f ${CORAL_ARTIFACTS}/${cname}${job_end}/latest)
+    comp_name=${cname}
+  fi
+  if [ -n "$latest" ]; then
+    #These can be used to override the "last good version"
+    #when there are breaking changes.   Jenkins will use
+    #$GOOD_* only if it is newer than the last version
+    #used by master
+    good_varname=GOOD_${upper_name}
+    good_version=${CORAL_ARTIFACTS}/${name}${job_end}/$version
+    declare ${good_varname}=${good_version}
+    blessed_varname=SL_${upper_name}_PREFIX
+    for subdir in /${comp_name}/TESTING/ /TESTING/scripts/ /${comp_name}/ /; do
+      vars=${latest}${subdir}.build_vars.sh
+      if [ -f ${vars} ]; then
+        break
+      fi
+    done
+    blessed_num=0
+    if [ ! -f ${vars} ]; then
+      print_status "$vars does not exist.   Check <build_vars> option"
+    else
+      source ${vars}
+      if [ -n "${!blessed_varname}" ]; then
+        blessed_num=$(basename $(dirname ${!blessed_varname}))
+      fi
+    fi
+    if [ $version -gt $blessed_num ]; then
+      if [ ! -d ${!good_varname} ]; then
+        print_status "${!good_varname} not found. Using latest instead"
+        if [ -n "${!blessed_varname}" ]; then
+          declare $upper_name=$(dirname ${!blessed_varname})
+        else
+          # No update-scratch job to get latest from.  Try component latest
+          print_status "No blessed version.  Using ${name} latest"
+          comp_latest=${CORAL_ARTIFACTS}/${name}${job_end}/latest
+          declare $upper_name=${comp_latest}
+        fi
+      else
+        declare $upper_name=${!good_varname}
+      fi
+    else
+      declare $upper_name=$(dirname ${!blessed_varname})
+    fi
+  fi
+  if [ "${DOCKER_IMAGE}x" == "x" ]; then
+    PREBUILT_AREA=${PREBUILT_AREA}${!upper_name}:
+  else
+    comp_src=`readlink -f "${!upper_name}"`
+    # If the upper_name is from an older build, it will be the
+    # container absolute path which readlink can not resolve
+    # in the Jenkins workspace
+    if [ "${comp_src}x" == "x" ]; then
+      comp_src=${!upper_name}
+    fi
+    comp_no=`basename "${comp_src}"`
+    # comp_target is path inside the container only
+    comp_target="${DIST_MOUNT}/${name}/${comp_no}"
+    PREBUILT_AREA="${PREBUILT_AREA}${comp_target}:"
+    # WORK_TARGET is base of where comp_target is in the Jenkins workspace
+    mkdir -p ${WORK_TARGET}/${name}/${comp_no}
```

Comment:
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]
(lint) Double quote to prevent globbing and word splitting. [SC2086]